lib/internal/test_runner/test.js

```diff
@@ -586,10 +586,13 @@ class Test extends AsyncResource {
         return;
       }
 
+      this.#abortController.abort();
```

Comment:
I think this moving this to `finally` would be better, I think there is no real nead for the signal to be aborted before running the hooks. @benjamingr WDYT?

Reply:
that won't necessarily work since `controller.abort()` is synchronous and cleanup performed on `abort` might not finish before the hook runs.